Nos/Views/Discover/DiscoverGrid.swift

```diff
@@ -33,15 +43,29 @@ struct DiscoverGrid: View {
                 Group {
                     switch searchController.state {
                     case .noQuery:
-                        StaggeredGrid(list: events, columns: columns, spacing: 15) { note in
-                            NoteButton(note: note, style: .golden)
-                                .padding(.bottom, 5)
-                                .matchedGeometryEffect(id: note.identifier, in: animation)
-                                .id(note.id)
+                        ScrollView {
+                            LazyVStack {
+                                ForEach(authors) { author in
+                                    AuthorCard(author: author) {
+                                        router.push(author)
+                                    }
+                                    .padding(.horizontal, 13)
+                                    .padding(.top, 5)
+                                    .readabilityPadding()
+                                    .task {
+                                        // TODO: optimize. Probably only needed once per author.
+                                        subscriptions[author.id] =
+                                            await relayService.requestMetadata(
+                                                for: author.hexadecimalPublicKey,
+                                                since: author.lastUpdatedMetadata
+                                            )
```

Comment:
Pretty much, yes! I would use `.onAppear` instead of `.task` because `.onAppear` only runs once, and `.task` is run every time the view is init, which in this case might happen every time a property on `author` changes.

The `relayService` also dedupes requests, so even if you did call `relayService.requestMetadata(...)` a bunch of times for the same author only one request would actually go out to relays. Also for this sort of request where we only want 1 event the relayService will close the request as soon as we get an event, or after a timeout, so we aren't leaving all these subscriptions open indefinitely.

Reply:
Cool, that makes me feel better about this. I checked the docs for [onAppear](https://developer.apple.com/documentation/swiftui/view/onappear(perform:)) and [task](https://developer.apple.com/documentation/swiftui/view/task(priority:_:)), and they both seem to run the same way. I also added both and logged them in a test run in the simulator, and both were called the same number of times, but I might be missing something.